src/platform/backends/lxd/lxd_virtual_machine_factory.cpp

```diff
@@ -67,7 +68,28 @@ void mp::LXDVirtualMachineFactory::prepare_instance_image(const mp::VMImage& /*
 
 void mp::LXDVirtualMachineFactory::hypervisor_health_check()
 {
-    auto reply = lxd_request(manager.get(), "GET", base_url);
+    QJsonObject reply;
+
+    try
+    {
+        reply = lxd_request(manager.get(), "GET", base_url);
+    }
+    catch (const LocalSocketConnectionException& e)
+    {
+        auto error_code = e.get_error();
+
+        if (error_code == QLocalSocket::ConnectionRefusedError)
+        {
+            throw std::runtime_error(
+                fmt::format("{}\nPlease ensure the LXD snap interface is connected via `snap connect "
+                            "multipass:lxd`\nand that the LXD snap is enabled and running.",
+                            e.what()));
+        }
+        else if (error_code == QLocalSocket::ServerNotFoundError)
+        {
+            throw std::runtime_error(fmt::format("{}\nPlease ensure the LXD snap is installed and running.", e.what()));
+        }
```

Comment:
I wonder if we should differentiate the message here? Sure, we can say what the underlying problem was, but the "solution", maybe we can just say the same across the board - "make sure the LXD snap is installed and that the `lxd` interface is connected" (sprinkle words to taste)?

As for "running"… LXD is socket-activated, so the socket should always be there, as long as it's installed. But maybe that's going into too much tech detail.

Reply:
Well, we can detect connection refused as opposed to the socket just not existing, so I thought it would be better to give a user a better clue as to how to fix their issue if there is permission denied, ie, the interface is not connected.  Also, if the LXD snap is disabled, it still leaves the socket hanging around, but nothing listening on it, so we'll get connection refused for that as well hence the "enabled and running" part of that message.

Server not found mainly implies LXD is not installed hence the shorter help message.

Reply:
Yeah, the problem is that it may well say the interface isn't connected after they install.

I just want to avoid this:
```plain
$ multipass launch
Please ensure LXD installed…
$ snap install lxd
$ multipass launch
Please ensure LXD interface connected…
```
:rage:

If it's uninstalled, let's tell them to install, run and connect; If it's not running, tell them to run and connect… You catch my drift :)

Reply:
Ah, ok, I do catch your drift.  So yeah, One Message to Rule Them All!

Reply:
Ok, there is now only one message for all socket connection failure types.  Note that the actual error string is prepended so there is still a clue about what the failure is.